src/librustc_trans/collector.rs

```diff
@@ -1019,7 +1019,9 @@ fn create_fn_trans_item<'a, 'tcx>(scx: &SharedCrateContext<'a, 'tcx>,
     let concrete_substs = monomorphize::apply_param_substs(scx,
                                                            param_substs,
                                                            &fn_substs);
-    assert!(concrete_substs.is_normalized_for_trans());
+    //assert!(concrete_substs.is_normalized_for_trans(),
+    //        "concrete_substs not normalized for trans: {:?}",
+    //        concrete_substs);
```

Comment:
Still, I have to admit: @nikomatsakis  it does not seem like you are ever actually reading the newly added `HAS_NORMALIZABLE_PROJECTION` bit; you are merely setting it...? Did you intend to incorporate that bit into this assertion, and then you changed your mind and decided to just remove the assertion entirely?